src/advisors/dep_check.ts

```diff
@@ -145,26 +150,63 @@ class DepCheckAdvisor implements IAdvisor {
                     }
                   )
                 );
+                // only resolved packages that are ported by
+                // emscripten and the compiler and linker flags for
+                // this package are added can be added into
+                // resolvedPkgs
+                resolvedPkgs.push(pkgName);
               } else {
                 nonPorts.push(m.groups.pkg);
               }
+            } else {
+              const n = lines[i].match(libraryReg);
+              const o = lines[i].match(includeDirReg);
+
+              // check if this line is related to resolved packages
+              // ${XXX_LIBRARIES} & ${XXX_INCLUDE_DIRS} are defined
+              // from statement `find_package`, also should be
+              // checked and resolved
+              const pkgName =
+                n && n[1] !== null
+                  ? n[1].toLowerCase().replace(/_/g, "")
+                  : o && o[1] !== null
+                  ? o[1].toLowerCase().replace(/_/g, "")
+                  : "";
+
+              // if this library has been resolved
+              if (pkgName !== "" && resolvedPkgs.includes(pkgName)) {
+                // comment this statement
+                actions.push(
+                  new FileChangeAction(
+                    proj.fileChangeManager,
+                    `Comment unnecessary statements for \`${pkgName}\` from \`CMakeLists.txt\` at line **${
+                      i + 1
+                    }**`,
+                    new FileRegion(file, i + 1),
+                    "# " + lines[i]
```

Comment:
If I have a below statement in cmake, according to the logic, line 3 & line 4 will be commented out. However, the end bracket `)` of `include_directories()` will also be commented out together, leading to syntax error. Shall we also consider such scenario together to make it right?

```cmake
1  include_directories(
2      ${CMAKE_SOURCE_DIR}/include
3      ${SDL2_INCLUDE_DIRS}
4      ${SDL2_TTF_INCLUDE_DIRS})
```

Reply:
Exactly, working on it